### PR TITLE
docs(messaging): update example

### PR DESCRIPTION
**Pull request by ahuounan**
### Summary

Fixes inaccurate documentation concerning usage and return type of messaging().requestPermission.

### Checklist

- [x] Supports `Android`
- [x] Supports `iOS`
- [ ] N/A - `e2e` tests added or updated in packages/\*\*/e2e
- [ ] N/A - Flow types updated
- [ ] N/A - Typescript types updated

### Test Plan
Not applicable

### Release Plan
Not applicable

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

File: docs/messaging/usage/index.md [hunk 1]
```diff
@@ -49,13 +49,16 @@ iOS prevents messages containing notification (or 'alert') payloads from being d
 This module provides a `requestPermission` method which triggers a native permission dialog requesting the user's permission:
 
 ```js
-import messaging from '@react-native-firebase/messaging';
+import messaging, { FirebaseMessagingTypes } from '@react-native-firebase/messaging';
```
Comment (Ehesp) [inline docs/messaging/usage/index.md:52]: This is TS only... i think we could change it to:

```
import messaging, { AuthorizationStatus } from '@react-native-firebase/messaging';
```

**Comment by ahuounan** [comment]
@Salakar signed the CLA

**Comment by ottob** [comment]
This new documentation did not work for me in my typescript project. Using this import:
```ts
import messaging, {
  AuthorizationStatus,
} from '@react-native-firebase/messaging'
```
I get this error:
> Module '"@react-native-firebase/messaging"' has no exported member 'AuthorizationStatus'. Did you mean to use 'import AuthorizationStatus from "@react-native-firebase/messaging"' instead? ts(2614)

If I instead use:
```ts
import messaging from '@react-native-firebase/messaging'

const enabled =
        authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
        authStatus === messaging.AuthorizationStatus.PROVISIONAL
```
It works as expected. And this matches [the comment](https://github.com/invertase/react-native-firebase/blob/5cda0ddf3cc4e62c253d3d190a6cb089bf2ccc83/packages/messaging/lib/index.d.ts#L435). Am I doing something wrong (Im a typescript noob) or should the docs be fixed?

**Comment by Ehesp** [comment]
cc @russellwheatley @dackers86 looks like it's not being exported - can you take a look please

**Pull request merged**